### PR TITLE
Commands used:

### PR DESCRIPTION
**Pull request by rwgk**
hgd pybind11_next
iwyu --nosafe_headers third_party/pybind11:pybind11
files=$(hg status | cut -c3-)
sed -i 's:#include "third_party/pybind11/include/pybind11/:#include "pybind11/:' $files

### CODE REVIEW

File: include/pybind11/attr.h [hunk 1]
```diff
@@ -8,9 +8,36 @@
...
+#include "pybind11/functional.h"
+#include "pybind11/iostream.h"
+#include "pybind11/numpy.h"
+#include "pybind11/operators.h"
+#include "pybind11/pybind11.h"
+#include "pybind11/pytypes.h"
+#include "pybind11/stl.h"
+#include "pybind11/stl_bind.h"
 
-#include "cast.h"
```
Comment (henryiii) [inline include/pybind11/attr.h:13]: I'd undo this change, I believe.

File: include/pybind11/attr.h [hunk 2]
```diff
@@ -8,9 +8,36 @@
     BSD-style license that can be found in the LICENSE file.
 */
 
-#pragma once
+#include <Python.h>
```
Comment (henryiii) [inline include/pybind11/attr.h:11]: Nice, it's keeping Python.h at the top. :)

File: include/pybind11/attr.h [hunk 3]
```diff
@@ -8,9 +8,36 @@
...
+#include <stdint.h>
+#include <string.h>
```
Comment (henryiii) [inline include/pybind11/attr.h:13]: I rather expect this is not needed.

File: include/pybind11/operators.h [hunk 4]
```diff
@@ -7,9 +7,18 @@
...
+#include <type_traits>
+#include <utility>
 
-#include "pybind11.h"
+#include "pybind11/attr.h"
+#include "pybind11/detail/common.h"
+#include "pybind11/pybind11.h"
+#include "pybind11/stl_bind.h"
+
+#pragma once
```
Comment (bstaletic) [inline include/pybind11/operators.h:21]: What's the point of moving `#pragma once` *after* all the include directives? Seems like a waste of I/O operations.

File: include/pybind11/pybind11.h [hunk 5]
```diff
@@ -8,6 +8,48 @@
...
+#include <new>
+#include <string>
+#include <type_traits>
+#include <typeindex>
+#include <typeinfo>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+#include <version>
```
Comment (bstaletic) [inline include/pybind11/pybind11.h:29]: This header doesn't exist before C++20.

File: include/pybind11/attr.h (hunk 1 above)
Comment (rwgk) [reply include/pybind11/attr.h:13]: It put in longer paths (`third_party/pybind11/include/pybind11`), I used `sed` to cut out the prefix but decided to keep the `pybind11` (having "explicit is better than implicit" in my head). I could cut out `pybind11` as well. BUT, I think the main question is, is the list of includes it produces useful, for cross-checking and/or occasional manual fixing up?

From this quick experiment I feel the tool is a lot too far away from what we'd need for fully automatic processing.

**Comment by rwgk** [comment]
Closing this PR (as planned; it was just meant to provide an easy means to show the diffs and enable commenting).

File: include/pybind11/operators.h (hunk 4 above)
Comment (henryiii) [reply include/pybind11/operators.h:21]: Doesn't really make a very big difference, but I have no idea why it moved it down.
